File: app/src/main/java/de/garten/training/depotflow/data/api/DepotApiClient.java
package de.garten.training.depotflow.data.api;

import java.util.List;

import de.garten.training.depotflow.core.ResultCallback;
import de.garten.training.depotflow.data.api.dto.OutboxDto;
import de.garten.training.depotflow.data.api.dto.SyncResponseDto;
import de.garten.training.depotflow.data.api.dto.WorkOrderDto;

public interface DepotApiClient {

    void loadWorkOrders(String depotId, String changedSince, ResultCallback<List<WorkOrderDto>> callback);

    void loadWorkOrder(String serverId, ResultCallback<WorkOrderDto> callback);

    void completeWorkOrder(String serverId, OutboxDto payload, ResultCallback<WorkOrderDto> callback);

    void pushOutbox(List<OutboxDto> payload, ResultCallback<SyncResponseDto> callback);
}

File: app/src/main/java/de/garten/training/depotflow/data/api/DepotApiClient.kt
package de.garten.training.depotflow.data.api

import de.garten.training.depotflow.core.ResultCallback
import de.garten.training.depotflow.data.api.dto.OutboxDto
import de.garten.training.depotflow.data.api.dto.SyncResponseDto
import de.garten.training.depotflow.data.api.dto.WorkOrderDto

interface DepotApiClient {
    fun loadWorkOrders(
        depotId: String,
        changedSince: String?,
        callback: ResultCallback<List<WorkOrderDto>>
    )

    fun loadWorkOrder(serverId: String, callback: ResultCallback<WorkOrderDto>)

    fun completeWorkOrder(
        serverId: String,
        payload: OutboxDto,
        callback: ResultCallback<WorkOrderDto>
    )

    fun pushOutbox(payload: List<OutboxDto>, callback: ResultCallback<SyncResponseDto>)
}
